twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -314,6 +318,31 @@ scrooge_scala_library = rule(
     provides = [DefaultInfo, ScroogeInfo, JavaInfo],
 )
 
+def _create_scala_struct(ctx):
+  """Create a scala provider in the shape expected by the IntelliJ bazel plugin."""
+  output_jars = []
+
+  if ctx.attr.exports:
+    for exp in ctx.attr.exports:
+      for j in exp[JavaInfo].outputs.jars:
+        output_jars.append(
+            struct(
+                class_jar = j.class_jar,
+                ijar = None,
+                source_jar = None,
+                source_jars = []))
+
+  for dep in ctx.attr.deps:
```

Comment:
okay, I guess this is correct, since the deps are actually the thrifts that we are going to trigger generation of, so pretending that this target created them, I guess is okay (rather than the aspect generating them).

Reply:
Having the aspects generate a Java/Scala provider is an interesting idea. I tried implementing this, but it doesn't seem like aspects support returning an old style struct provider. I tried returning a struct with a `scala` field from the scrooge aspect, but it wasn't visible in the IJ plugin. 

Once the `java_common.merge` fix is released, the scrooge aspect could attach a JavaInfo with multiple outputs to the `thrift_library` targets.

Reply:
always a yak stack....

next year in Jerusalem.